### ISSUE TITLE
Transaction kernel proof demo - `wasm-bindgen` extension

### ISSUE THREAD

**Issue by frisitano** [0xPolygonMiden/examples#143]
To support our ambitions of generating client side transaction proofs as discussed in #142 we must extend the `wasm-bindgen` API to introduce functionality for performing transaction proving.  This will be done by integrating the `miden-tx` crate. 

I am thinking that we can have a two stage process - prepare transaction and then prove transaction. In the prepare transaction phase we fetch data and compile the transaction program.  In the prove transaction phase we take the prepared transaction and prove it.  To support this we need to introduce two additional endpoints:

- prepare_transaction: this endpoint will be responsible for preparing a transaction and returning the `PreparedTransaction` object to the client.  The client can then display the data from the `PreparedTransaction` object.

```rust
#[wasm_bindgen]
pub prepare_transaction() -> Result<PreparedTransaction, JsValue> {
    ...
}
```

- prove_transaction: this endpoint will be responsible for proving the `PreparedTransaction` and returning a `ProvenTransaction` object.  The client can then display the `ProvenTransaction` information. 

```rust
#[wasm_bindgen]
pub fn prove_transaction() -> Result<ProvenTransaction, JsValue> {
   ....
}
```

**Comment by Dominik1999** [comment]
That looks good, thanks. I think for the purpose of the demo; we will have a super simple webpage with two buttons. One for each endpoint. 

I will start working on the frontend now.

**Comment by frisitano** [comment]
> That looks good, thanks. Can you add a `verify_transaction()` endpoint?
> 
> Then we can have three buttons and a super simple webpage.

Yes good call - I will add it.

**Comment by frisitano** [comment]
closed by #145
